pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -58,15 +109,13 @@ func (m *migration) getNotificationChannelMap() (channelsPerOrg, defaultChannels
 	allChannelsMap := make(channelsPerOrg)
 	defaultChannelsMap := make(defaultChannelsPerOrg)
 	for i, c := range allChannels {
-		if _, ok := allChannelsMap[c.OrgID]; !ok { // new seen org
-			allChannelsMap[c.OrgID] = make(map[interface{}]*notificationChannel)
-		}
-		if c.Uid != "" {
-			allChannelsMap[c.OrgID][c.Uid] = &allChannels[i]
-		}
-		if c.ID != 0 {
-			allChannelsMap[c.OrgID][c.ID] = &allChannels[i]
+		if c.Type == "hipchat" || c.Type == "sensu" {
+			m.mg.Logger.Error("alert migration error: discontinued notification channel found", "type", c.Type, "name", c.Name, "uid", c.Uid)
```

Comment:
While not as good as having a single log message, there is a warning log downstream if any alerts were attached to a discontinued channel (I'll make it more descriptive).

```go
	// Filter receiver names.
	filteredReceiverNames := make(map[string]interface{})
	for _, uidOrId := range channelIDs {
		recv, ok := receivers[uidOrId]
		if ok {
			filteredReceiverNames[recv.Name] = struct{}{} // Deduplicate on contact point name.
		} else {
			m.mg.Logger.Warn("ignoring obsolete notification channel", "uid", uidOrId)
		}
	}
```

On a related note though, I don't think we should be just dropping discontinued channels on migration. IMO it would be better to create a "stub" contact point with the same or similar name that they can fill out at their leisure. I'll create an issue to discuss separate to this PR.